src/generate.js

```diff
@@ -143,18 +144,20 @@ export const generateCSS = (
     stringHandlers /* : StringHandlers */ = {},
     useImportant /* : boolean */ = true
 ) /* : string */ => {
-    const merged = styleTypes.reduce(recursiveMerge);
+    const merged /* : OrderedElements */ = styleTypes.reduce(
+        recursiveMerge,
+        new OrderedElements());
 
-    const plainDeclarations = {};
+    const plainDeclarations = new OrderedElements();
     let generatedStyles = "";
 
-    Object.keys(merged).forEach(key => {
+    merged.forEach((key, val) => {
```

Comment:
unrelated: Although I generally think stuff like `forEach` is best, regular for loops are faster which might matter in performance-sensitive applications. It might be worth looking into optimizing Aphrodite in this way at some point.